src/runtime_src/core/pcie/tools/xbmgmt/xspi.cpp

```diff
@@ -2180,3 +2224,127 @@ int XSPI_Flasher::upgradeFirmware2Drv(std::istream& mcsStream0,
     // Disable bitstream guard.
     return removeBitstreamGuard(mDev.get(), mFlashDev, bsGuardAddr);
 }
+
+const std::string magic = "XRTDATA";
+struct flash_data_header {
+    char magic[7] = "";
```

Comment:
Is this value meant to be a "null" terminated string?  or are you simply attempting to initialize the structure to known values.  

If it is to be a "null" terminated string, then add room to include the null character.

If you are attempting to initialize the structure to a known value, then use following syntax when creating the structure (as shown later below)
`flash_data_header header = {0};`

Reply:
will use the = { 0 }  syntax.